web/src/components/countryhistorytitle.jsx

```diff
@@ -0,0 +1,26 @@
+import { useSelector } from 'react-redux';
+import styled from 'styled-components';
+import { useTranslation } from '../helpers/translation';
+import React from 'react';
+import { formatTimeRange } from '../helpers/formatting';
+
+const Text = styled.span`
+  font-size: 1.1em;
+`;
+
+export const CountryHistoryTitle = ({ translationKey }) => {
+  const { __, i18n } = useTranslation();
+  const aggregation = useSelector((state) => state.application.selectedTimeAggregate);
+  const localExists = i18n.exists(`${translationKey}.${aggregation}`, { fallbackLng: i18n.language });
+  const localDefaultExists = i18n.exists(`${translationKey}.default`, { fallbackLng: i18n.language });
+  /*
+  Use local for aggregation if exists, otherwise use local default if exists. If no translation exists, use english
+  */
+  return (
+    <Text>
+      {localExists
+        ? __(`${translationKey}.${aggregation}`)
+        : __(`${translationKey}.default`, formatTimeRange(localDefaultExists ? i18n.language : 'en', aggregation))}
```

Comment:
I tried using translateIfExist but it didn't cover my exact use case is it would still fallback to english.

I agree it's quite a lot to have in the render. My thought process here was that the component is so tiny that it was OK to have more complex functionality in the render.

Reply:
In general I think it's still better to move the complex functionality out and give it proper space and naming (instead of working within the constraints inside JSX), but as mentioned it's a nitpick in this case :)